autometa/common/external/samtools.py

```diff
@@ -81,11 +43,33 @@ def sort(sam, bam, nproc=mp.cpu_count()):
         </path/to/output/alignment.bam>
     nproc : int, optional
         Number of processors to be used. By default uses all the processors of the system 
-    """
     
-    cmd = f'samtools view -@{nproc} -bS {sam} | samtools sort -@{nproc} -o {bam}'
-    run(cmd,bam)
+    Raises
+    ------
+    ChildProcessError
+        Samtools did not run successfully, returns retcode.
+    
+    """
 
+    cmd = f'samtools view -@{nproc} -bS {sam} | samtools sort -@{nproc} -o {bam}'
+    logger.debug(f'cmd: {cmd}')
+    samtools_out_dir = os.path.dirname(bam)
+    tempdir = tempfile.mkdtemp(suffix=None, prefix='samtools', dir=samtools_out_dir)
+    os.chdir(tempdir)
```

Comment:
Is changing to the samtools temp dir necessary here?

Reply:
I am also unclear why you need the temp directory. If nothing else it is weird to change to that directory and then later to delete it. Not sure what the OS will do in that situation, and also other scripts might assume that the directory they are in doesn't change (although I guess it is good practice not to assume that sort of thing and use full paths where possible).